File: srss-api/user-microservice/src/main/java/ru/application/usermicroservice/enums/SquadsSortedType.java
package ru.application.usermicroservice.enums;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.experimental.FieldDefaults;

@Getter
@RequiredArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
public enum SquadsSortedType {

    RATING_ASC("score ASC"),

    RATING_DESC("score DESC"),

    NUMBER_USERS_ASC("number_users ASC"),

    NUMBER_USERS_DESC("number_users DESC"),

    ALPHABET("name ASC");

    String value;
}

File: srss-frontend/src/app/enums/SquadsSortedType.ts
export enum SquadsSortedType {
  RATING_ASC = 'RATING_ASC',
  RATING_DESC = 'RATING_DESC',
  NUMBER_USERS_ASC = 'NUMBER_USERS_ASC',
  NUMBER_USERS_DESC = 'NUMBER_USERS_DESC',
  ALPHABET = 'ALPHABET'
}


export const SQUADS_SORTED_TYPE_LABELS: Record<SquadsSortedType, string> = {
  [SquadsSortedType.RATING_ASC]: 'рейтингу ↑',
  [SquadsSortedType.RATING_DESC]: 'рейтингу ↓',
  [SquadsSortedType.NUMBER_USERS_ASC]: 'участникам ↑',
  [SquadsSortedType.NUMBER_USERS_DESC]: 'участникам ↓',
  [SquadsSortedType.ALPHABET]: 'названию'
};
